core/src/mast/serialization/mod.rs

```diff
@@ -1,30 +1,63 @@
-use alloc::vec::Vec;
-
+//! The serialization format of MastForest is as follows:
+//!
+//! (Metadata)
+//! - MAGIC
+//! - VERSION
+//!
+//! (lengths)
+//! - decorators length (`usize`)
+//! - nodes length (`usize`)
+//!
+//! (procedure roots)
+//! - procedure roots (`Vec<MastNodeId>`)
+//!
+//! (raw data)
+//! - Decorator data
+//! - Node data
+//! - String table
+//!
+//! (info structs)
+//! - decorator infos (`Vec<DecoratorInfo>`)
+//! - MAST node infos (`Vec<MastNodeInfo>`)
+//!
+//! (Decorator ID maps)
+//! - before enter map (`BTreeMap<MastNodeId, Vec<DecoratorId>>`)
+//! - after exit map (`BTreeMap<MastNodeId, Vec<DecoratorId>>`)
```

Comment:
Not for this PR, but I wonder if there is a way to refactor this so that all decorator-related data is in one section. This way, if we wanted to drop decorators from the serialized binary, we could just truncate this section without doing any extra processing.

Let's create an issue for this.

Reply:
#1489